docs/client/request.md

```diff
@@ -211,6 +211,37 @@ Header method returns header value via key, this method will visit all field in
 func (r *Request) Header(key string) []string
 ```
 
+### Headers
+
+Headers returns all headers in the request using an iterator. You can use `maps.Collect()` to collect all headers into a map.
+The returned value is valid until the request object is released. Any future calls to Headers method will return the modified value. Do not store references to returned value. Make copies instead.
+
+```go title="Signature"
+func (r *Request) Headers() iter.Seq2[string, []string]
+```
+
+```go title="Example"
+req := client.AcquireRequest()
+
+req.AddHeader("Golang", "Fiber")
+req.AddHeader("Test", "123456")
+req.AddHeader("Test", "654321")
+
+for k, v := range req.Headers() {
+  fmt.Printf("Header Key: %s, Header Value: %v\n", k, v)
+}
+```
+
+<details>
+<summary>Click here to see the result</summary>
```

Comment:
Remove this details/summary, and just show the results in the docs.

Reply:
@ReneWerner87 what do you think

Reply:
please put the complete example in the block(details) and display the result directly below it

the idea is that everywhere in the docs the examples can be folded out because they can become larger and we do not take the focus of the consumers with it

like here
https://github.com/gofiber/fiber/blob/main/docs/whats_new.md?plain=1#L457-L491
https://github.com/gofiber/fiber/blob/main/docs/whats_new.md?plain=1#L721-L755
...